tree/ntuple/v7/inc/ROOT/RNTupleWriteOptions.hxx

```diff
@@ -42,6 +42,136 @@ public:
+</td>
+</tr>
+
+<tr>
+<td>`MaxUnzippedClusterSize`</td>
+<td>`std::size_t`</td>
+<td>1280 MiB</td>
+<td>
+Memory limit for committing a cluster: with very high compression ratio, we need a limit
+on how large the I/O buffer can grow during writing.
+</td>
+</tr>
+
+<tr>
+<td>`InitialUnzippedPageSize`</td>
+<td>`std::size_t`</td>
+<td>256</td>
+<td>
+Initially, columns start with a page of this size. The default value is chosen to accomodate at least 32 elements
+of 64 bits, or 64 elements of 32 bits. If more elements are needed, pages are increased up until the byte limit
+given by the option `MaxUnzippedPageSize` or until the total page buffer limit is reached (as a sum of all page buffers).
+The total write buffer limit needs to be large enough to hold the initial pages of all columns.
+</td>
+</tr>
+
+<tr>
+<td>`MaxUnzippedPageSize`</td>
+<td>`std::size_t`</td>
+<td>1 MiB</td>
+<td>
+Pages can grow only to the given limit in bytes.
+</td>
+</tr>
+
+<tr>
+<td>`PageBufferBudget`</td>
+<td>`std::size_t`</td>
+<td>0 / auto</td>
+<td>
+The maximum size that the sum of all page buffers used for writing into a persistent sink are allowed to use.
+If set to zero, RNTuple will auto-adjust the budget based on the value of `ApproxZippedClusterSize`.
+If set manually, the size needs to be large enough to hold all initial page buffers.
+The total amount of memory for writing is larger, e.g. for the additional compressed buffers etc.
+Use RNTupleModel::EstimateWriteMemoryUsage() for the total estimated memory use for writing.
+The default values are tuned for a total write memory of around 400 MiB per fill context.
+</td>
+</tr>
+
+<tr>
+<td>`UseBufferedWrite`</td>
+<td>`bool`</td>
+<td>`true`</td>
+<td>
+Whether to use buffered writing (with RPageSinkBuf). This buffers compressed pages in memory, reorders them
+to keep pages of the same column adjacent, and coalesces the writes when committing a cluster.
+</td>
+</tr>
+
+<tr>
+<td>`UseDirectIO`</td>
+<td>`bool`</td>
+<td>`false`</td>
+<td>
+Whether to use Direct I/O for writing. Note that this introduces alignment requirements that may very between
+filesystems and platforms.
+</td>
+</tr>
+
+<tr>
+<td>`WriteBufferSize`</td>
+<td>`std::size_t`</td>
+<td>4 MiB</td>
+<td>
+Buffer size to use for writing to files, must be a multiple of 4096 bytes. Testing suggests that 4MiB gives best
+performance (with Direct I/O) at a reasonable memory consumption.
+</td>
+</tr>
+
+<tr>
+<td>`UseImplicitMT`</td>
+<td>EImplicitMT</td>
+<td>EImplicitMT::kDefault</td>
+<td>
+Whether to use implicit multi-threading to compress pages. Only has an effect if buffered writing is turned on.
+</td>
+</tr>
+
+<tr>
+<td>`EnablePageChecksums`</td>
+<td>`bool`</td>
+<td>`true`</td>
+<td>
+If set, checksums will be calculated and written for every page.
+</td>
+</tr>
+
+<tr>
+<td>`EnableSamePageMerging`</td>
+<td>`bool`</td>
+<td>`true`</td>
+<td>
+If set, identical pages are deduplicated and aliased on disk. Requires page checksums.
```

Comment:
IIRC, when this option is enabled, page checksums are enabled implictly as well, right? If so, this should probably be mentioned as well.

Reply:
By default, both are turned on. If page checksums are disabled, it will automatically turn off same page merging. If same page merging is attempted to be turned on and page checksums are off, it will throw. I added two sentences on this.